scripts/enableTrading.ts

```diff
@@ -0,0 +1,184 @@
+import Contracts from '../components/Contracts';
+import { DeployedContracts, execute, getNamedSigners, InstanceName } from '../utils/Deploy';
+import Logger from '../utils/Logger';
+import { DEFAULT_DECIMALS, TokenSymbol } from '../utils/TokenData';
+import '@nomiclabs/hardhat-ethers';
+import '@typechain/hardhat';
+import { CoinGeckoClient } from 'coingecko-api-v3';
+import Decimal from 'decimal.js';
+
+interface EnvOptions {
+    ENABLE_TRADING?: boolean;
+}
+
+const { ENABLE_TRADING: enableTrading }: EnvOptions = process.env as any as EnvOptions;
+
+interface TokenData {
+    address: string;
+    ethVirtualBalance: Decimal;
+    tokenVirtualBalance: Decimal;
+}
+
+const MAX_PRECISION = 16;
+
+const main = async () => {
+    const { deployer } = await getNamedSigners();
+    const carbonPOL = await DeployedContracts.CarbonPOL.deployed();
+
+    const tokenAddressesEnv = process.env.TOKEN_ADDRESSES;
+    if (!tokenAddressesEnv) {
+        console.error(`no tokens passed in - pass in token addresses like so: TOKEN_ADDRESSES='0x...','0x...'`);
+        return;
+    }
+
+    // Remove single quotes and whitespace, then split by commas
+    const allTokens = tokenAddressesEnv
+        .replace(/'/g, '')
+        .split(',')
+        .map((address) => address.trim());
```

Comment:
Suggestion: store allTokens as `Set<string>` and normalize the addresses by calling ethers' `getAddress` (or lower/upper case the address). This way, you will automatically ignore duplicates.

Reply:
removed that code altogether